src/Events/Custom_Tables/V1/Models/Builder.php

```diff
@@ -553,8 +565,46 @@ public function delete() {
 	 */
 	public function find( $value, $column = null ) {
 		$column = null === $column ? $this->model->primary_key_name() : $column;
+		$conf   = tribe( Configuration::class );
+
+		// Memoize disabled?
+		if ( $conf->get( 'TEC_NO_MEMOIZE_CT1_MODELS' ) ) {
+			return $this->where( $column, $value )->first();
+		}
+
+		// Check if we memoized this instance.
+		$key    = self::generate_cache_key( $this->model, $column, $value );
+		$result = tribe_cache()->get( $key, Tribe__Cache_Listener::TRIGGER_SAVE_POST, null, Tribe__Cache::NON_PERSISTENT );
+
+		if ( $result ) {
+			return $result;
+		}
+
+		// Not memoized, fetch it.
+		$result = $this->where( $column, $value )->first();
+		if ( $result ) {
+			// Store on model so we can use it to cache bust later.
+			$result->cache_key = $key;
+
+			tribe_cache()->set( $key, $result, Tribe__Cache::NON_PERSISTENT, Tribe__Cache_Listener::TRIGGER_SAVE_POST );
```

Comment:
@lucatume I considered the same thing. The reason I decided to do memoizing because I wasn't confident of no edge case bugs with persistent cache - perhaps I'm being too cautious? I did find at least one place that would potentially cause a bug, and we would need a way to clear the cache as a group (which I don't think we have an API for?) because these events are not being updated via `wp_insert_post`.

Perhaps kick cache to a later update?